.github/workflows/tox-pytest.yml

```diff
@@ -59,13 +62,22 @@ jobs:
 
       - name: Set default gcp credentials
         id: gcloud-auth
+        continue-on-error: true
         uses: "google-github-actions/auth@v1"
         with:
-          credentials_json: "${{ secrets.TOX_PYTEST_SA_KEY }}"
+          workload_identity_provider: "projects/345950277072/locations/global/workloadIdentityPools/gh-actions-pool/providers/gh-actions-provider"
+          service_account: "tox-pytest-github-action@catalyst-cooperative-pudl.iam.gserviceaccount.com"
+          create_credentials_file: true
 
-      - name: Run PyTest with Tox
+      - name: Run PyTest with Tox against GCS cache (internal PRs)
         run: |
           tox -- --gcs-cache-path gs://zenodo-cache.catalyst.coop
+        if: ${{ steps.gcloud-auth.outcome == 'success' }}
+
+      - name: Run PyTest with Tox against Zenodo (external PRs)
+        run: |
+          tox
+        if: ${{ steps.gcloud-auth.outcome == 'failure' }}
 
```

Comment:
Smart workaround! I guess we could have used this work around and continued to use service account keys?

Reply:
We could have! And if we want to, we can split these two PRs if that feels better. I ran into this *after* I got the WIF stuff working, though, which is why it's all in one PR.

Reply:
Ok! No need to split the PR. Thanks for getting WIF working!